Allow to modify memory settings for Postgres and Solr and bump to 2GB each by default

yvanzo opened the pull request:
The two additional configuration parameters for Postgres were hard-coded in a script installed in the Docker image for the service `db`. It was not easy to change `shared_buffers`’ value to meet meet resources and satisfy needs.

This patch drops the script and set configuration parameters from the `command` defined for the `db` service. It allows to change these values without having to rebuild images, and to add any other configuration parameter or option: just override `command` in a local compose file.

It also increases the value for `shared_buffers` to `2048MB` by default, which is more suitable when running a mirror to speed web service up. It does the same with the Java heap for Solr.

Finally, it documents these new defaults in installation steps and recommends to adjust memory settings to address speed concerns, with a full example of creating a local compose override file.

References:
* https://hub.docker.com/_/postgres/
* https://www.postgresql.org/docs/9.5/app-postgres.html
* https://github.com/apache/lucene-solr/blob/releases/lucene-solr/7.7.2/solr/bin/solr.in.sh/

yvanzo committed:
Set Postgres configuration parameters from command

The two additional configuration parameters were hardcoded in a script
installed in the Docker image for the service 'db'. It was not easy to
change 'shared_buffers' value to meet meet resources and satisfy needs.

This patch drops the script and set configuration parameters from the
'command' defined for the db service. It allows to change these values
without having to rebuild images, and to add any other configuration
parameter or option: just override 'command' in a local compose file.

It also increases the value for shared_buffers to 2048MB by default,
which is more suitable when running a mirror to speed web service up.

References:
* https://hub.docker.com/_/postgres/
* https://www.postgresql.org/docs/9.5/app-postgres.html

mayhem reviewed (changes requested):
Looks nice, thank you.

Two requests:

1. Please include custom-db-options.yml as you suggest in IRC as an example for how to do this.
2. Please update the docs to strongly urge users to set this setting in order to get decent performance.

yvanzo committed:
Increase the Java heap for Solr to 2 GB as well

mayhem commented on README.md:
````diff
@@ -327,6 +327,36 @@ admin/configure add publishing-all-ports
 sudo docker-compose up -d
 ```
 
+#### Modify memory settings
````
Great! How about adding "NOTE: You should consider modifying your memory settings in order to give your database a sufficient amount of ram, otherwise your database could run very slowly. See section Modify memory settings." to the #create-database?

yvanzo replied:
Should be good now. I did the same for Solr. See https://github.com/metabrainz/musicbrainz-docker/pull/148/commits/273c65ee2d16106baa8ef1a28ddaed02008ea812, https://github.com/metabrainz/musicbrainz-docker/pull/148/commits/b298cbe3a42b3d1a604c554e119fd37097cc693e, and https://github.com/metabrainz/musicbrainz-docker/pull/148/commits/8821ad4c2c2db3a53b40de3bb76cb9cad63cebef.

yvanzo committed:
Document how to bump memory to 4 GB for Pg & Solr

This also provides a full example of how to customize Docker Compose
configuration by creating a custom override file under `local/compose/`.

yvanzo committed:
Document default memory settings for installation

Recommend to modify memory settings to address speed concerns

mayhem reviewed (approved):
THANK YOU!
